packages/core/src/utils/websocket.ts

```diff
@@ -107,6 +132,21 @@ export class RelayerWebsocket {
   // | Private API |
   // ---------------
 
+  private request(message: SubscriptionMessage | UnsubscriptionMessage): void {
+    if (
+      this.ws?.readyState === this.ws?.CLOSED ||
+      this.ws?.readyState === this.ws?.CLOSING
+    ) {
```

Comment:
wdyt about another separate case for when the state is `CONNECTING`, namely waiting for the state to change before sending the request?

Reply:
i think it would be more appropriate to encode reconnection logic in a module that manages raw WebSocket connections like this one. then, raw WebSocket connections can be thought of as either in an "open" or explicit "fail/failing" state, and the higher level manager can control reconnects. will do this in a future PR.